### PR TITLE
#AIR-908 Security fix for kube-rbac-proxy v0.8

### PR DESCRIPTION
**Pull request by ShaunakJoshi1407**
trivy vulnerability check: 
```
$ trivy fs .
2023-03-28T13:59:12.502+0530	INFO	Vulnerability scanning is enabled
2023-03-28T13:59:12.504+0530	INFO	Secret scanning is enabled
2023-03-28T13:59:12.504+0530	INFO	If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-03-28T13:59:12.505+0530	INFO	Please see also https://aquasecurity.github.io/trivy/v0.37/docs/secret/scanning/#recommendation for faster secret detection
2023-03-28T13:59:12.596+0530	INFO	Number of language-specific files: 1
2023-03-28T13:59:12.596+0530	INFO	Detecting gomod vulnerabilities...

go.mod (gomod)

Total: 1 (UNKNOWN: 0, LOW: 0, MEDIUM: 1, HIGH: 0, CRITICAL: 0)

┌──────────────────┬───────────────┬──────────┬───────────────────┬────────────────┬──────────────────────────────────────────────────────────┐
│     Library      │ Vulnerability │ Severity │ Installed Version │ Fixed Version  │                          Title                           │
├──────────────────┼───────────────┼──────────┼───────────────────┼────────────────┼──────────────────────────────────────────────────────────┤
│ k8s.io/client-go │ CVE-2020-8565 │ MEDIUM   │ 0.19.2            │ 0.20.0-alpha.2 │ kubernetes: Incomplete fix for CVE-2019-11250 allows for │
│                  │               │          │                   │                │ token leak in logs when...                               │
│                  │               │          │                   │                │ https://avd.aquasec.com/nvd/cve-2020-8565                │
```

Output of `make build`:
```
$ make build
>> building for darwin/amd64 to _output/kube-rbac-proxy-darwin-amd64
GOARCH=amd64 \
	GOOS=darwin \
	CGO_ENABLED=0 \
	go build --installsuffix cgo -o _output/kube-rbac-proxy-darwin-amd64 github.com/brancz/kube-rbac-proxy
cp _output/kube-rbac-proxy-darwin-amd64 _output/kube-rbac-proxy
```

Output of `make pf9-image` :
```
$ make pf9-image 
Building Docker image for pf9 Repo...
Sending build context to Docker daemon    379MB
Step 1/6 : FROM gcr.io/distroless/static:nonroot
 ---> b152689d931f
Step 2/6 : ARG BINARY=kube-rbac-proxy-linux-amd64
 ---> Using cache
 ---> 8bf32848b332
Step 3/6 : COPY _output/$BINARY /usr/local/bin/kube-rbac-proxy
 ---> 8c2023ca9a79
Step 4/6 : EXPOSE 8080
 ---> Running in 48fd9a96cba6
Removing intermediate container 48fd9a96cba6
 ---> b03ba71e31eb
Step 5/6 : USER 65532:65532
 ---> Running in 3d2e43ef4ba5
Removing intermediate container 3d2e43ef4ba5
 ---> 5d62956142c5
Step 6/6 : ENTRYPOINT ["/usr/local/bin/kube-rbac-proxy"]
 ---> Running in 45eb609c3b53
Removing intermediate container 45eb609c3b53
 ---> df86de946d90
Successfully built df86de946d90
Successfully tagged platform9/kube-rbac-proxy:v0.8.0-pmk-

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
echo docker.io/platform9/kube-rbac-proxy:v0.8.0-pmk- > /Users/shaunakj/go/src/github.com/platform9/kube-rbac-proxy/container-tag
```

TC Build: 
https://teamcity.platform9.horse/buildConfiguration/Pf9project_Platform9ComponentsForKubernetes_KubeRbacProxy/2624525

#### Manual Testing:
Edited the luigi-controller-manager deployment in a 1.20 cluster to point to the image that was built by TC above. The pod is up and running with this new image.

luigi-controller-manager deployment:
```
kubectl get deployment luigi-controller-manager -n luigi-system -oyaml
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "2"
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"apps/v1","kind":"Deployment","metadata":{"annotations":{},"labels":{"control-plane":"controller-manager"},"name":"luigi-controller-manager","namespace":"luigi-system"},"spec":{"replicas":1,"selector":{"matchLabels":{"control-plane":"controller-manager"}},"template":{"metadata":{"labels":{"control-plane":"controller-manager"}},"spec":{"containers":[{"args":["--secure-listen-address=0.0.0.0:8443","--upstream=http://127.0.0.1:8080/","--logtostderr=true","--v=10"],"image":"localhost:5100/kubebuilder/kube-rbac-proxy:v0.8.0","name":"kube-rbac-proxy","ports":[{"containerPort":8443,"name":"https"}]},{"args":["--metrics-addr=127.0.0.1:8080","--enable-leader-election"],"command":["/manager"],"image":"localhost:5100/platform9/luigi-plugins:v0.3","imagePullPolicy":"IfNotPresent","name":"manager","resources":{"limits":{"cpu":"100m","memory":"125Mi"},"requests":{"cpu":"20m","memory":"50Mi"}}}],"terminationGracePeriodSeconds":10}}}}
  creationTimestamp: "2023-04-13T06:43:10Z"
  generation: 2
  labels:
    control-plane: controller-manager
  managedFields:
  - apiVersion: apps/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:kubectl.kubernetes.io/last-applied-configuration: {}
        f:labels:
          .: {}
          f:control-plane: {}
      f:spec:
        f:progressDeadlineSeconds: {}
        f:replicas: {}
        f:revisionHistoryLimit: {}
        f:selector: {}
        f:strategy:
          f:rollingUpdate:
            .: {}
            f:maxSurge: {}
            f:maxUnavailable: {}
          f:type: {}
        f:template:
          f:metadata:
            f:labels:
              .: {}
              f:control-plane: {}
          f:spec:
            f:containers:
              k:{"name":"kube-rbac-proxy"}:
                .: {}
                f:args: {}
                f:imagePullPolicy: {}
                f:name: {}
                f:ports:
                  .: {}
                  k:{"containerPort":8443,"protocol":"TCP"}:
                    .: {}
                    f:containerPort: {}
                    f:name: {}
                    f:protocol: {}
                f:resources: {}
                f:terminationMessagePath: {}
                f:terminationMessagePolicy: {}
              k:{"name":"manager"}:
                .: {}
                f:args: {}
                f:command: {}
                f:image: {}
                f:imagePullPolicy: {}
                f:name: {}
                f:resources:
                  .: {}
                  f:limits:
                    .: {}
                    f:cpu: {}
                    f:memory: {}
                  f:requests:
                    .: {}
                    f:cpu: {}
                    f:memory: {}
                f:terminationMessagePath: {}
                f:terminationMessagePolicy: {}
            f:dnsPolicy: {}
            f:restartPolicy: {}
            f:schedulerName: {}
            f:securityContext: {}
            f:terminationGracePeriodSeconds: {}
    manager: kubectl-client-side-apply
    operation: Update
    time: "2023-04-13T06:43:10Z"
  - apiVersion: apps/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:spec:
        f:template:
          f:spec:
            f:containers:
              k:{"name":"kube-rbac-proxy"}:
                f:image: {}
    manager: kubectl-edit
    operation: Update
    time: "2023-04-13T06:55:21Z"
  - apiVersion: apps/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          f:deployment.kubernetes.io/revision: {}
      f:status:
        f:availableReplicas: {}
        f:conditions:
          .: {}
          k:{"type":"Available"}:
            .: {}
            f:lastTransitionTime: {}
            f:lastUpdateTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
            f:type: {}
          k:{"type":"Progressing"}:
            .: {}
            f:lastTransitionTime: {}
            f:lastUpdateTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
            f:type: {}
        f:observedGeneration: {}
        f:readyReplicas: {}
        f:replicas: {}
        f:updatedReplicas: {}
    manager: kube-controller-manager
    operation: Update
    time: "2023-04-13T06:55:27Z"
  name: luigi-controller-manager
  namespace: luigi-system
  resourceVersion: "3182"
  uid: cd550c2c-bffc-403d-8a18-11e60e0b7cd8
spec:
  progressDeadlineSeconds: 600
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      control-plane: controller-manager
  strategy:
    rollingUpdate:
      maxSurge: 25%
      maxUnavailable: 25%
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        control-plane: controller-manager
    spec:
      containers:
      - args:
        - --secure-listen-address=0.0.0.0:8443
        - --upstream=http://127.0.0.1:8080/
        - --logtostderr=true
        - --v=10
        image: docker.io/platform9/kube-rbac-proxy:v0.8.0-pmk-2624525
        imagePullPolicy: IfNotPresent
        name: kube-rbac-proxy
        ports:
        - containerPort: 8443
          name: https
          protocol: TCP
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
      - args:
        - --metrics-addr=127.0.0.1:8080
        - --enable-leader-election
        command:
        - /manager
        image: localhost:5100/platform9/luigi-plugins:v0.3
        imagePullPolicy: IfNotPresent
        name: manager
        resources:
          limits:
            cpu: 100m
            memory: 125Mi
          requests:
            cpu: 20m
            memory: 50Mi
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 10
status:
  availableReplicas: 1
  conditions:
  - lastTransitionTime: "2023-04-13T06:44:06Z"
    lastUpdateTime: "2023-04-13T06:44:06Z"
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
  - lastTransitionTime: "2023-04-13T06:43:10Z"
    lastUpdateTime: "2023-04-13T06:55:27Z"
    message: ReplicaSet "luigi-controller-manager-696469b58d" has successfully progressed.
    reason: NewReplicaSetAvailable
    status: "True"
    type: Progressing
  observedGeneration: 2
  readyReplicas: 1
  replicas: 1
  updatedReplicas: 1
```

luigi-controller-manager-pod-status:

```
kubectl get pod luigi-controller-manager-696469b58d-n6ghv -n luigi-system
NAME                                        READY   STATUS    RESTARTS   AGE
luigi-controller-manager-696469b58d-n6ghv   2/2     Running   0          11m
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,17 +1,71 @@
...
 	github.com/oklog/run v1.0.0
 	github.com/spf13/pflag v1.0.5
-	golang.org/x/net v0.0.0-20200707034311-ab3426394381
-	gopkg.in/yaml.v2 v2.2.8
+	golang.org/x/net v0.7.0
+	gopkg.in/yaml.v2 v2.3.0
 	k8s.io/api v0.19.2
 	k8s.io/apimachinery v0.19.2
 	k8s.io/apiserver v0.19.2
 	k8s.io/client-go v0.19.2
```
Comment (cruizen) [inline go.mod:14]: The k8s APIs can be upgraded to v0.20.* (like [0.20.15](https://pkg.go.dev/k8s.io/client-go@v0.20.15)) to fix the one remaining MEDIUM vuln - though we don't have to. The rest looks good. 👍

Comment (ShaunakJoshi1407) [reply go.mod:14]: So are we done here atleast from high and critical vulnerabilities standpoint ?

File: Makefile [hunk 2]
```diff
@@ -3,6 +3,21 @@ all: check-license build generate test
 GO111MODULE=on
 export GO111MODULE
 
+BUILDDIR=$(CURDIR)
+registry_url ?= docker.io
```
Comment (mithilarun) [inline Makefile]: Why is this necessary? Do you expect to set this variable from outside? If so - It needs to use an environment variable. If the environment variable isn't set, default to docker.io.

File: Makefile [hunk 3]
```diff
@@ -3,6 +3,21 @@ all: check-license build generate test
 GO111MODULE=on
 export GO111MODULE
 
+BUILDDIR=$(CURDIR)
```
Comment (mithilarun) [inline Makefile]: Why is this needed? Why not use CURDIR directly?

Comment (ShaunakJoshi1407) [reply Makefile]: Done

File: Makefile (hunk 2 above)
Comment (ShaunakJoshi1407) [reply Makefile]: Fixed

**Review by cruizen — APPROVED** [review]
/lgtm
/approve

File: go.mod (hunk 1 above)
Comment (cruizen) [reply go.mod:14]: Obviously. If there are 0 CRIT and 0 HIGH vulns, ...

**Review by cruizen — CHANGES_REQUESTED** [review]
Remove the large binary files from the PR changeset

**Review by cruizen — APPROVED** [review]

**Pull request merged**